avocado/core/job.py

```diff
@@ -494,25 +450,17 @@ def _log_job_debug_info(self, variants):
         self._log_job_id()
 
     def create_test_suite(self):
-        """
-        Creates the test suite for this Job
-
-        This is a public Job API as part of the documented Job phases
-        """
-        refs = self.config.get('run.references')
-        ignore_missing = self.config.get('run.ignore_missing_references')
-        self._make_test_suite(refs, ignore_missing)
-        if not self.test_suite:
-            if refs:
-                e_msg = ("No tests found for given test references, try "
-                         "'avocado list -V %s' for details") % " ".join(refs)
-            else:
-                e_msg = ("No test references provided nor any other arguments "
-                         "resolved into tests. Please double check the "
-                         "executed command.")
-            raise exceptions.JobTestSuiteEmptyError(e_msg)
-
-        self.result.tests_total = len(self.test_suite)
+        try:
+            self.test_suite = TestSuite.from_config(self.config,
+                                                    runner=self._test_runner_name)
```

Comment:
You're usually looking for a smaller number of arguments, so I have to ask: any reason for letting `from_config` itself find out what the runner is?  IMO, this would minimize the interface changes once the current runner is dropped, and hopefully we'll only have one runner (the "next" runner, with multiple spawners).

Reply:
There is no reason. :) You are right, makes sense. Will change on a v2.